File: android/app/src/main/java/app/gymbuddy/ui/components/AppIcon.kt
package app.gymbuddy.ui.components

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.ArrowBack
import androidx.compose.material.icons.automirrored.outlined.ArrowForward
import androidx.compose.material.icons.automirrored.outlined.Chat
import androidx.compose.material.icons.automirrored.outlined.List
import androidx.compose.material.icons.automirrored.outlined.Send
import androidx.compose.material.icons.filled.Restaurant
import androidx.compose.material.icons.filled.Bolt
import androidx.compose.material.icons.filled.BarChart
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.DarkMode
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FavoriteBorder
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Image
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material.icons.filled.KeyboardArrowLeft
import androidx.compose.material.icons.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Language
import androidx.compose.material.icons.filled.LightMode
import androidx.compose.material.icons.filled.LocalFireDepartment
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.MilitaryTech
import androidx.compose.material.icons.filled.MoreHoriz
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Pause
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Replay
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.Search
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.Tune
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.outlined.AccessTime
import androidx.compose.material.icons.outlined.AutoAwesome
import androidx.compose.material.icons.outlined.CalendarMonth
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material.icons.outlined.People
import androidx.compose.material.icons.outlined.HelpOutline
import androidx.compose.material.icons.outlined.Public
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Project-wide icon names map to Material Icons.
 *
 * Names mirror those used in the design tokens (home, dumbbell, flame, …)
 * so calling code stays close to the JSX prototype:
 *
 * ```kotlin
 * AppIcon(name = "heart-fill", size = 22.dp, tint = ...)
 * ```
 */
@Composable
fun AppIcon(
    name: String,
    modifier: Modifier = Modifier,
    size: Dp = 22.dp,
    tint: Color = Color.Unspecified,
) {
    val vector: ImageVector = when (name) {
        "home" -> Icons.Filled.Home
        "dumbbell" -> Icons.Filled.FitnessCenter
        "flame" -> Icons.Filled.LocalFireDepartment
        "spark" -> Icons.Outlined.AutoAwesome
        "heart" -> Icons.Filled.FavoriteBorder
        "heart-fill" -> Icons.Filled.Favorite
        "user" -> Icons.Outlined.Person
        "users" -> Icons.Outlined.People
        "cog" -> Icons.Filled.Settings
        "plus" -> Icons.Filled.Add
        "check" -> Icons.Filled.Check
        "close" -> Icons.Filled.Close
        "chevron-right" -> Icons.Filled.KeyboardArrowRight
        "chevron-left" -> Icons.Filled.KeyboardArrowLeft
        "chevron-down" -> Icons.Filled.KeyboardArrowDown
        "arrow-right" -> Icons.AutoMirrored.Outlined.ArrowForward
        "arrow-left" -> Icons.AutoMirrored.Outlined.ArrowBack
        "search" -> Icons.Filled.Search
        "calendar" -> Icons.Outlined.CalendarMonth
        "bell" -> Icons.Filled.Notifications
        "filter" -> Icons.Filled.Tune
        "bolt" -> Icons.Filled.Bolt
        "rewind" -> Icons.Filled.Replay
        "star" -> Icons.Filled.Star
        "send" -> Icons.AutoMirrored.Outlined.Send
        "apple" -> Icons.Filled.Restaurant
        "medal" -> Icons.Filled.MilitaryTech
        "trophy" -> Icons.Filled.EmojiEvents
        "chart" -> Icons.Filled.BarChart
        "play" -> Icons.Filled.PlayArrow
        "pause" -> Icons.Filled.Pause
        "image" -> Icons.Filled.Image
        "camera" -> Icons.Filled.CameraAlt
        "edit" -> Icons.Filled.Edit
        "language" -> Icons.Filled.Language
        "moon" -> Icons.Filled.DarkMode
        "sun" -> Icons.Filled.LightMode
        "location" -> Icons.Filled.LocationOn
        "clock" -> Icons.Outlined.AccessTime
        "list" -> Icons.AutoMirrored.Outlined.List
        "grid" -> Icons.Filled.GridView
        "more" -> Icons.Filled.MoreHoriz
        "globe" -> Icons.Outlined.Public
        "sliders" -> Icons.Filled.Tune
        "chat" -> Icons.AutoMirrored.Outlined.Chat
        "schedule" -> Icons.Filled.Schedule
        else -> Icons.Outlined.HelpOutline
    }
    Box(modifier = modifier.size(size)) {
        Icon(imageVector = vector, contentDescription = name, tint = tint, modifier = Modifier.size(size))
    }
}

File: ios/GymBuddy/GymBuddy/Components/AppIcon.swift
import SwiftUI

/// SF Symbols mapping for the icon names used in the prototype.
public enum AppIconName: String {
    case home, dumbbell, flame, spark, heart, heartFill = "heart-fill"
    case user, users, cog, plus, check, close
    case chevronRight = "chevron-right"
    case chevronLeft = "chevron-left"
    case chevronDown = "chevron-down"
    case arrowRight = "arrow-right"
    case arrowLeft = "arrow-left"
    case search, calendar, bell, filter, bolt, rewind, star, send
    case apple, medal, trophy, chart, play, pause, image, camera
    case edit, language, moon, sun, location, clock, list, grid, more
    case globe, sliders

    public var sfSymbol: String {
        switch self {
        case .home: return "house.fill"
        case .dumbbell: return "dumbbell.fill"
        case .flame: return "flame.fill"
        case .spark: return "sparkles"
        case .heart: return "heart"
        case .heartFill: return "heart.fill"
        case .user: return "person.fill"
        case .users: return "person.2.fill"
        case .cog: return "gearshape.fill"
        case .plus: return "plus"
        case .check: return "checkmark"
        case .close: return "xmark"
        case .chevronRight: return "chevron.right"
        case .chevronLeft: return "chevron.left"
        case .chevronDown: return "chevron.down"
        case .arrowRight: return "arrow.right"
        case .arrowLeft: return "arrow.left"
        case .search: return "magnifyingglass"
        case .calendar: return "calendar"
        case .bell: return "bell.fill"
        case .filter: return "line.3.horizontal.decrease.circle"
        case .bolt: return "bolt.fill"
        case .rewind: return "arrow.uturn.backward"
        case .star: return "star.fill"
        case .send: return "paperplane.fill"
        case .apple: return "applelogo"
        case .medal: return "medal.fill"
        case .trophy: return "trophy.fill"
        case .chart: return "chart.bar.fill"
        case .play: return "play.fill"
        case .pause: return "pause.fill"
        case .image: return "photo"
        case .camera: return "camera.fill"
        case .edit: return "pencil"
        case .language: return "globe"
        case .moon: return "moon.fill"
        case .sun: return "sun.max.fill"
        case .location: return "location.fill"
        case .clock: return "clock"
        case .list: return "list.bullet"
        case .grid: return "square.grid.2x2.fill"
        case .more: return "ellipsis"
        case .globe: return "globe"
        case .sliders: return "slider.horizontal.3"
        }
    }
}

public struct AppIcon: View {
    public let name: AppIconName
    public var size: CGFloat = 22
    public var color: Color = .primary

    public init(_ name: AppIconName, size: CGFloat = 22, color: Color = .primary) {
        self.name = name; self.size = size; self.color = color
    }

    public var body: some View {
        Image(systemName: name.sfSymbol)
            .font(.system(size: size * 0.85, weight: .semibold))
            .foregroundStyle(color)
            .frame(width: size, height: size)
    }
}
